### ISSUE TITLE
torch._dynamo.exc.Unsupported: ObservedKeyError exception running Gguf llama model in vLLM

### ISSUE THREAD

**Issue by bnellnm** [pytorch/pytorch#136502]
### 🐛 Describe the bug

Running the `TheBloke/TinyLlama-1.1B-Chat-v1.0-GGUF` model with gguf quantization leads to a `ObservedKeyError` in dynamo.

Reproduction steps:

Save to `bug.py`
```
import torch
import vllm

from huggingface_hub import hf_hub_download
from vllm import LLM, SamplingParams

# Sample prompts.                                                                                                                 
prompts = [
    "Hello, my name is",
    "The president of the United States is",
    "The capital of France is",
    "The future of AI is",
]

# Create a sampling params object.                                                                                                
sampling_params = SamplingParams(temperature=0.8, top_p=0.95)

model=hf_hub_download("TheBloke/TinyLlama-1.1B-Chat-v1.0-GGUF",
                      filename="tinyllama-1.1b-chat-v1.0.Q4_K_M.gguf")

llm = LLM(model=model, enforce_eager=True)

outputs = llm.generate(prompts, sampling_params)
# Print the outputs.                                                                                                              
for output in outputs:
    prompt = output.prompt
    generated_text = output.outputs[0].text
    print(f"Prompt: {prompt!r}, Generated text: {generated_text!r}")
```

```
export VLLM_TEST_DYNAMO_GRAPH_CAPTURE=1
export VLLM_TEST_DYNAMO_FULLGRAPH_CAPTURE=1
python3 bug.py
```

I'm using the 2.5 nightly version of pytorch and the 0.6.1.post2 (b05f5c923) version of vLLM.
Note: this was masked by a different issue in pytorch 2.4

Partial Stacktrace:
```
../pt24/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py:111: in call_function
    return tx.inline_user_function_return(self, [*self.self_args(), *args], kwargs)
../pt24/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:836: in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
../pt24/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:3011: in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
../pt24/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:3139: in inline_call_
    tracer.run()
../pt24/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:983: in run
    while self.step():
../pt24/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:898: in step
    self.exception_handler(e)
../pt24/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:1496: in exception_handler
    raise raised_exception
../pt24/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:895: in step
    self.dispatch_table[inst.opcode](self, inst)
../pt24/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:582: in wrapper
    return inner_fn(self, inst)
../pt24/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:301: in impl
    self.push(fn_var.call_function(self, self.popn(nargs), {}))
../pt24/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py:967: in call_function
    return handler(tx, args, kwargs)
../pt24/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py:848: in builtin_dispatch
    rv = fn(tx, args, kwargs)
../pt24/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py:766: in call_self_handler
    result = self_handler(tx, *args, **kwargs)
../pt24/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py:1472: in call_getitem
    return args[0].call_method(tx, "__getitem__", args[1:], kwargs)
../pt24/lib/python3.10/site-packages/torch/_dynamo/variables/dicts.py:261: in call_method
    return self.getitem_const_raise_exception_if_absent(tx, args[0])
../pt24/lib/python3.10/site-packages/torch/_dynamo/variables/dicts.py:224: in getitem_const_raise_exception_if_absent
    raise_observed_exception(KeyError, tx, self)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

e = <class 'KeyError'>, tx = <torch._dynamo.symbolic_convert.InliningInstructionTranslator object at 0x76f3d82b5390>
vt = ConstDictVariable()

    def raise_observed_exception(e, tx, vt):
        from .variables import BuiltinVariable

        # CPython here raises an exception. Since there is no python code, we have to manually setup the exception
        # stack and raise the exception.
        exception_vt = BuiltinVariable(e).call_function(vt, [], {})
        tx.exn_vt_stack.append(exception_vt)
>       raise observed_exception_map[e]
E       torch._dynamo.exc.ObservedKeyError

../pt24/lib/python3.10/site-packages/torch/_dynamo/exc.py:234: ObservedKeyError
```

### Versions

```
Collecting environment information...
PyTorch version: 2.5.0+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.30.1
Libc version: glibc-2.35

Python version: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-6.5.0-35-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 12.5.82
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration:
GPU 0: NVIDIA H100 80GB HBM3
GPU 1: NVIDIA H100 80GB HBM3
GPU 2: NVIDIA H100 80GB HBM3
GPU 3: NVIDIA H100 80GB HBM3
GPU 4: NVIDIA H100 80GB HBM3
GPU 5: NVIDIA H100 80GB HBM3
GPU 6: NVIDIA H100 80GB HBM3
GPU 7: NVIDIA H100 80GB HBM3

Nvidia driver version: 555.42.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture: x86_64
CPU op-mode(s): 32-bit, 64-bit
Address sizes: 46 bits physical, 57 bits virtual
Byte Order: Little Endian
CPU(s): 128
On-line CPU(s) list: 0-127
Vendor ID: GenuineIntel
Model name: Intel(R) Xeon(R) Platinum 8462Y+
CPU family: 6
Model: 143
Thread(s) per core: 2
Core(s) per socket: 32
Socket(s): 2
Stepping: 8
CPU max MHz: 4100.0000
CPU min MHz: 800.0000
BogoMIPS: 5600.00
Flags: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hfi vnmi avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr ibt amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
Virtualization: VT-x
L1d cache: 3 MiB (64 instances)
L1i cache: 2 MiB (64 instances)
L2 cache: 128 MiB (64 instances)
L3 cache: 120 MiB (2 instances)
NUMA node(s): 2
NUMA node0 CPU(s): 0-31,64-95
NUMA node1 CPU(s): 32-63,96-127
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit: Not affected
Vulnerability L1tf: Not affected
Vulnerability Mds: Not affected
Vulnerability Meltdown: Not affected
Vulnerability Mmio stale data: Not affected
Vulnerability Retbleed: Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1: Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2: Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds: Not affected
Vulnerability Tsx async abort: Not affected

Versions of relevant libraries:
[pip3] flashinfer==0.0.9+cu121torch2.3
[pip3] mypy==1.9.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.4
[pip3] onnx==1.14.1
[pip3] onnxruntime==1.18.1
[pip3] pytorch-triton==3.1.0+5fe38ffd73
[pip3] torch==2.5.0+cu124
[pip3] torchaudio==2.5.0.dev20240919+cu121
[pip3] torchvision==0.20.0.dev20240919+cu121
[pip3] triton==3.0.0
[conda] Could not collect
```

cc @ezyang @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @amjames @rec @zou3519

**Comment by exclamaforte** [comment]
The repro seems to work now, feel free to reopen if it's still broken.
```
Prompt: 'Hello, my name is', Generated text: " John and I've been a computer user for years. I've always"
Prompt: 'The president of the United States is', Generated text: ' not a king. King George VI of Great Britain served as the first monarch to'
Prompt: 'The capital of France is', Generated text: ' Paris, which is known for its art, cuisine, and history.'
Prompt: 'The future of AI is', Generated text: ' here: Manufacturing robots can now self-organize and improve at'
[rank0]:[W325 17:26:28.071867840 ProcessGroupNCCL.cpp:1496] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
```
